### PR TITLE
ZTP option (DHCPv6) and SZTP (DHCPv4 and v6) support

### PR DESCRIPTION
**Pull request by eamonjd**
Summary
---------
- Added ZTP DHCPv6 option BOOTFILE_URL support (previously only equivalent DHCPv4 option was supported) (RFC5970)
- Added SZTP option support for both DHCPv4 and DHCPv6 requests and responses. (RFC8572)
- Added DHCPv6 user-class  option (RFC8415) 
As the existing DHCPv4 and DHCPv6 tcpdump output are not consistent with each other, these changes follow this convention and adopts the respective style for DHCPv4 and DHCPv6. 

Details
-------
A) Rather than `Unknown (143)`, output now specifies `SZTP-Redirect (143)`
Added support for SZTP-Redirect Option 143 on requests
```text
15:26:16.092235 00:00:44:01:00:00 (oui Unknown) > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x10, ttl 128, id 0, offset 0, flags [none], proto UDP (17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: BOOTP/DHCP, Request from 00:00:44:01:00:00 (oui Unknown), length 300, xid 0xba761038, Flags [none]
          Client-Ethernet-Address 00:00:44:01:00:00 (oui Unknown)
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Discover
            Parameter-Request (55), length 10:
              MTU (26), Subnet-Mask (1), BR (28), Default-Gateway (3)
              Domain-Name (15), Domain-Name-Server (6), LOG (7), Hostname (12)
              Unknown (143), Classless-Static-Route (121)
            Vendor-Class (60), length 6: "Arista"
            Client-ID (61), length 6: "^@D^A^@^@"


15:25:54.957572 00:00:44:01:00:00 (oui Unknown) > Broadcast, ethertype IPv4 (0x0800), length 342: (tos 0x10, ttl 128, id 0, offset 0, flags [none], proto UDP
(17), length 328)
    0.0.0.0.bootpc > 255.255.255.255.bootps: BOOTP/DHCP, Request from 00:00:44:01:00:00 (oui Unknown), length 300, xid 0xd53905c, Flags [none]
          Client-Ethernet-Address 00:00:44:01:00:00 (oui Unknown)
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: Request
            Server-ID (54), length 4: 10.10.0.2
            Requested-IP (50), length 4: 10.10.0.4
            Parameter-Request (55), length 10:
              MTU (26), Subnet-Mask (1), BR (28), Default-Gateway (3)
              Domain-Name (15), Domain-Name-Server (6), LOG (7), Hostname (12)
              SZTP-Redirect (143), Classless-Static-Route (121)
            Vendor-Class (60), length 6: "Arista"
            Client-ID (61), length 6: "^@D^A^@^@"
```

B) Previously tcpdump outputted the binary values, it now shows the individual values in text format, `(T143 Option 143 -> SZTP-Redirect (143))`, e.g. 

```text
15:26:16.099544 IP (tos 0x10, ttl 128, id 0, offset 0, flags [none], proto UDP (17), length 412)
    10.10.0.2.bootps > 10.10.0.4.bootpc: BOOTP/DHCP, Reply, length 384, xid 0xba761038, Flags [none]
          Your-IP 10.10.0.4
          Client-Ethernet-Address 00:00:44:01:00:00 (oui Unknown)
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message Option 53, length 1: ACK
            Server-ID Option 54, length 4: 10.10.0.2
            Lease-Time Option 51, length 4: 28778
            MTU Option 26, length 2: 1500
            Subnet-Mask Option 1, length 4: 255.255.0.0
            Default-Gateway Option 3, length 4: 10.10.0.1
            Domain-Name Option 15, length 18: "aristanetworks.com"
            Domain-Name-Server Option 6, length 4: 10.10.0.1
            T143 Option 143, length 84: 1730676,1953526586,791638906,1953509678,1685026670,1819238756,778268525,1140852,1953526586,791621936,774975534,8083333
12,594048116,1886599727,794505776,808532532,943075386,876099120,976894008,943217978,942684208


15:26:16.099544 4e:ae:9c:94:66:0d (oui Unknown) > 00:00:44:01:00:00 (oui Unknown), ethertype IPv4 (0x0800), length 426: (tos 0x10, ttl 128, id 0, offset 0, fl
ags [none], proto UDP (17), length 412)
    10.10.0.2.bootps > 10.10.0.4.bootpc: BOOTP/DHCP, Reply, length 384, xid 0xba761038, Flags [none]
          Your-IP 10.10.0.4
          Client-Ethernet-Address 00:00:44:01:00:00 (oui Unknown)
          Vendor-rfc1048 Extensions
            Magic Cookie 0x63825363
            DHCP-Message (53), length 1: ACK
            Server-ID (54), length 4: 10.10.0.2
            Lease-Time (51), length 4: 28778
            MTU (26), length 2: 1500
            Subnet-Mask (1), length 4: 255.255.0.0
            Default-Gateway (3), length 4: 10.10.0.1
            Domain-Name (15), length 18: "aristanetworks.com"
            Domain-Name-Server (6), length 4: 10.10.0.1
            SZTP-Redirect (143), length 84:
              instance#1: "https://sztp1.download.com/", length 26
              instance#2: "https://10.10.0.0/", length 17
              instance#3: "[https://](https://github.com/the-tcpdump-group/tcpdump/compare/master...eamonjd:master)[2001:4860:4860::8888]:8080", length 35
```

C) In DHCPv6 solicit requests, rather than `opt_59`, now use `Bootfile-URL`, e.g.
```text
19:24:08.422125 IP6 (flowlabel 0x1de3e, hlim 1, next-header UDP (17) payload length: 199) fe80::2edd:e9ff:fec9:3288.dhcpv6-client > ff02::1:2.dhcpv6-server: \
[udp sum ok] dhcp6 solicit (xid=422a14 (vendor-specific-info) (client-ID hwaddr/time type 1 time 700062283 2cdde9c93288) (option-request opt_59 DNS-search-li\
st DNS-server) (elapsed-time 725) (user-class) (IA_NA IAID:3922277000 T1:3600 T2:5400))


19:24:08.422125 2c:dd:e9:c9:32:88 (oui Arista Networks) > 33:33:00:01:00:02 (oui Unknown), ethertype 802.1Q (0x8100), length 257: vlan 2204, p 0, ethertype IP
v6 (0x86dd), (flowlabel 0x1de3e, hlim 1, next-header UDP (17) payload length: 199) fe80::2edd:e9ff:fec9:3288.dhcpv6-client > ff02::1:2.dhcpv6-server: dhcp6 so
licit (xid=422a14 (vendor-specific-info) (client-ID hwaddr/time type 1 time 700062283 2cdde9c93288) (option-request Bootfile-URL DNS-search-list DNS-server) (
elapsed-time 725) (user-class) (IA_NA IAID:3922277000 T1:3600 T2:5400))
```

D) In DHCPv6 ZTP responses, rather than `(opt_59)`, now use `Bootfile-URL <value>`, e.g.

```text
15:15:33.148466 IP6 (flowlabel 0x1570c, hlim 64, next-header UDP (17) payload length: 157) fe80::40d3:61ff:fe62:3810.dhcpv6-server > fe80::200:1ff:fe01:0.dhcpv6-client: [udp sum ok] dhcp6 advertise (xid=654242 (IA_NA IAID:16842752 T1:0 T2:0 (IA_ADDR 1234:5678::4 pltime:18000 vltime:28800)) (client-ID hwaddr/time type 1 time 701792102 000001010000) (server-ID hwaddr/time type 1 time 701792110 42d361623810) (opt_59) (DNS-search-list aristanetworks.com.) (DNS-server 1234:5678::2))


15:15:33.148466 42:d3:61:62:38:10 (oui Unknown) > 00:00:01:01:00:00 (oui Unknown), ethertype IPv6 (0x86dd), length 211: (flowlabel 0x1570c, hlim 64, next-header UDP (17) payload length: 157) fe80::40d3:61ff:fe62:3810.dhcpv6-server > fe80::200:1ff:fe01:0.dhcpv6-client: dhcp6 advertise (xid=654242 (IA_NA IAID:16842752 T1:0 T2:0 (IA_ADDR 1234:5678::4 pltime:18000 vltime:28800)) (client-ID hwaddr/time type 1 time 701792102 000001010000) (server-ID hwaddr/time type 1 time 701792110 42d361623810) (Bootfile-URL my-startup-config) (DNS-search-list aristanetworks.com.) (DNS-server 1234:5678::2))
```

E) In DHCPv6 SZTP responses, rather than `(opt_136)`, now use `(SZTP-redirect *(uri))`, e.g.
 `(SZTP-redirect [https://sztp1.download.com,https://sztp2.download.com:8080,https://10.10.0.0,https://10.10.0.0:8000,https://](https://sztp1.download.com%2Chttps//sztp2.download.com:8080,https://10.10.0.0,https://10.10.0.0:8000,https://)[2001:4860:4860::8888]:8080)`

```text
14:35:46.638223 IP6 (flowlabel 0xe612c, hlim 64, next-header UDP (17) payload length: 281) fe80::cc0d:b4ff:fe8a:3384.dhcpv6-server > fe80::200:1ff:fe01:0.dhcpv6-client: [udp sum ok] dhcp6 reply (xid=5f98e6 (IA_NA IAID:16842752 T1:0 T2:0 (IA_ADDR 1234:5678::4 pltime:7200 vltime:28800)) (client-ID hwaddr/time type 1
time 701530515 000001010000) (server-ID hwaddr/time type 1 time 701530524 ce0db48a3384) (opt_136) (DNS-search-list aristanetworks.com.) (DNS-server 1234:5678::2))


13:35:46.638223 ce:0d:b4:8a:33:84 (oui Unknown) > 00:00:01:01:00:00 (oui Unknown), ethertype IPv6 (0x86dd), length 335: (flowlabel 0xe612c, hlim 64, next-header UDP (17) payload length: 281) fe80::cc0d:b4ff:fe8a:3384.dhcpv6-server > fe80::200:1ff:fe01:0.dhcpv6-client: dhcp6 reply (xid=5f98e6 (IA_NA IAID:16842752 T1:0 T2:0 (IA_ADDR 1234:5678::4 pltime:7200 vltime:28800)) (client-ID hwaddr/time type 1 time 701530515 000001010000) (server-ID hwaddr/time type 1 time 701530524 ce0db48a3384) (SZTP-redirect [https://sztp1.download.com,https://sztp2.download.com:8080,https://10.10.0.0,https://10.10.0.0:8000,https://](https://sztp1.download.com%2Chttps//sztp2.download.com:8080,https://10.10.0.0,https://10.10.0.0:8000,https://)[2001:4860:4860:\
:8888]:8080) (DNS-search-list aristanetworks.com.) (DNS-server 1234:5678::2))
```

F) Add support for DHCPv6 user-class field as per [RFC8415](https://datatracker.ietf.org/doc/html/rfc8415#section-21.15).

For example, now include the user-class values in the DHCPv6 output `(user-class Arista;Test-Dhcpv6Discover;4.21.XX)`
Previously:
```text
09:08:32.009292 IP6 (flowlabel 0x09903, hlim 1, next-header UDP (17) payload length: 138) fe80::200:44ff:fe01:0.dhcpv6-client > ff02::1:2.dhcpv6-server: [udp sum ok] dhcp6 inf-req (xid=b5fcf (vendor-specific-info) (client-ID hwaddr type 1 000044010000) (option-request opt_59 DNS-search-list DNS-server) (elapsed-time 331) (user-class))
```
Updated to:
```text
08:08:32.009292 IP6 (flowlabel 0x09903, hlim 1, next-header UDP (17) payload length: 138) fe80::200:44ff:fe01:0.546 > ff02::1:2.547: [udp sum ok] dhcp6 inf-req (xid=b5fcf (vendor-specific-info) (client-ID hwaddr type 1 000044010000) (option-request Bootfile-URL DNS-search-list DNS-server) (elapsed-time 331) (user-class Arista;Test-Dhcpv6Discover;4.21.XX))
```

### CODE REVIEW

**Review by its0x08 — APPROVED** [review]

**Comment by fenner** [comment]
I've made some minor whitespace changes in the code (to better match the surrounding style), moved one #define so that it's not in the middle of the NTP values, and squashed the two commits into one.  I've also updated the commit message, to have the protocols in the title and some more detail about the changes in the body.

its0x8 has reviewed the changes back in April and approved.

Is there anything else we can do to move this pull request towards being merged?

File: print-bootp.c [hunk 1]
```diff
@@ -993,6 +997,51 @@ rfc1048_print(netdissect_options *ndo,
...
+				 */
+				u_int suboptnumber = 1;
+
+				while (len >= 2) {
+					suboptlen = GET_BE_U_2(bp);
+					bp += 2;
+					len -= 2;
+					ND_PRINT("\n\t	    ");
+					ND_PRINT("instance#%u: ", suboptnumber);
+					if (suboptlen == 0) {
```
Comment (infrastation) [inline print-bootp.c:1019]: This check seems to be modelled after the `TAG_USER_CLASS` option above, which implements the requirement of RFC 3004 Section 4. However, RFC 8572 Section 8.3 does not explicitly require `uri-length` to be greater than zero. However, it does require `URI` to start with `https://`, so in any valid use case `suboptlen` would be greater than zero.

However, since this is a packet analyzer, which should be able to produce useful decoding of a packet even if the packet is malformed, it would make a bit more sense instead of skipping the rest of the option to print the empty URI and then to skip to the next URI within the same option if possible. Also it may be useful to print "invalid" or some such after every URI that does not begin with `https://`, which would also cover the empty string.

File: print-bootp.c [hunk 2]
```diff
@@ -993,6 +997,51 @@ rfc1048_print(netdissect_options *ndo,
...
+					if (len < suboptlen) {
+						ND_PRINT("[ERROR: invalid option]");
+						bp += len;
+						len = 0;
+						break;
+					}
+					ND_PRINT("\"");
+					nd_printjn(ndo, bp, suboptlen);
+					ND_PRINT("\"");
+					ND_PRINT(", length %u", suboptlen);
```
Comment (infrastation) [inline print-bootp.c:1034]: DHCP options typically print the length first in any case, and then either print an "invalid" marker or the option value, which makes it possible to see the cause of the problem when the length is invalid. It would be useful to follow the same pattern for the suboptions here.

File: print-dhcp6.c [hunk 3]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
 			ND_PRINT(")");
 			break;
 
+		case DH6OPT_BOOTFILE_URL:
+			if (optlen < 2) {
+				ND_PRINT(" ?)");
```
Comment (infrastation) [inline print-dhcp6.c:795]: Please make this message something that allows the user understand what is wrong.

File: print-dhcp6.c [hunk 4]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+				break;
+			}
+			tp = (const u_char *)(dh6o + 1);
+			ND_PRINT(" ");
+			nd_printjnp(ndo, tp, optlen);
```
Comment (infrastation) [inline print-dhcp6.c:800]: According to RFC 5970 Section 3.1, this option value is not NUL-terminated. If it is not supposed to be zero-padded either, `nd_printjn()` would be a better fit because it would make any out-of-place NUL characters apparent.

File: print-dhcp6.c [hunk 5]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+			tp = (const u_char *)(dh6o + 1);
+			first_list_value = TRUE;
+			remainder_len = optlen;
+			while (tp < cp + sizeof(*dh6o) + optlen - 2) {
+				if (first_list_value == FALSE) {
+					ND_PRINT(",");
+				}
+				first_list_value = FALSE;
+				subopt_len = GET_BE_U_2(tp);
+				if (subopt_len > remainder_len-2) {
```
Comment (infrastation) [inline print-dhcp6.c:816]: This would be a good place to print another "invalid" marker so the user can see something is wrong with the packet.

File: print-dhcp6.c [hunk 6]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+					break;
+				}
+				tp += 2;
+				nd_printjnp(ndo, tp, subopt_len);
```
Comment (infrastation) [inline print-dhcp6.c:820]: Please use `nd_printjn()` here too.

File: print-dhcp6.c [hunk 7]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+				tp += subopt_len;
+				remainder_len -= (subopt_len+2);
+			}
+			if (remainder_len != 0 ) {
+				ND_PRINT(" ?");
```
Comment (infrastation) [inline print-dhcp6.c:825]: This could be a better "invalid" marker too.

File: print-dhcp6.c [hunk 8]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+				break;
+			}
+			tp = (const u_char *)(dh6o + 1);
+			ND_PRINT(" ");
+			nd_printjnp(ndo, tp, optlen);
+			ND_PRINT(")");
+			break;
+
+		case DH6OPT_SZTP_REDIRECT:
+		case DH6OPT_USER_CLASS:
```
Comment (infrastation) [inline print-dhcp6.c:805]: Forgot to add: since `DH6OPT_USER_CLASS` appears both here and in the test, it would be useful to mention it in the commit message as well.

File: print-dhcp6.c (hunk 3 above)
Comment (fenner) [reply print-dhcp6.c:795]: There are 20 other instances of `ND_PRINT(" ?)");`, so it seemed like it was a known pattern in this file.  If you do want the new code to be inconsistent with the old, what about using `ND_PRINT(" [|opt])";`, to use the common truncation marker and try to evoke that it's the option that's truncated?

Comment (infrastation) [reply print-dhcp6.c:795]: That's a fair point. Alright, let's keep this error reporting consistent for now.

File: print-bootp.c (hunk 2 above)
Comment (fenner) [reply print-bootp.c:1034]: As you mentioned above, this was modeled after the implementation of DHCP_USER_CLASS:
https://github.com/the-tcpdump-group/tcpdump/blob/19c46646c07bc76e403856d18dc0611a96a1627d/print-bootp.c#L985-L988

It sounds like you'd rather see it modeled after AGENT_CIRCUIT's method of printing multiple items?
https://github.com/the-tcpdump-group/tcpdump/blob/19c46646c07bc76e403856d18dc0611a96a1627d/print-bootp.c#L870-L893

Current:
```
           SZTP-Redirect (143), length 141: 
                   instance#1: "https://sztp1.download.com", length 26
                   instance#2: "https://sztp2.download.com:8080", length 31
                   instance#3: "https://10.10.0.0", length 17
                   instance#4: "https://10.10.0.0:8000", length 22
                   instance#5: "https://[2001:4860:4860::8888]:8080", length 35
```

Proposed based on feedback:
```
           SZTP-Redirect (143), length 141: 
                   length 26: "https://sztp1.download.com"
                   length 31: "https://sztp2.download.com:8080"
                   length 17:: "https://10.10.0.0"
                   length 22: "https://10.10.0.0:8000"
                   length 35: "https://[2001:4860:4860::8888]:8080"
                   length 0: ""
                   length 999: length goes past end of option
```

File: print-dhcp6.c (hunk 7 above)
Comment (fenner) [reply print-dhcp6.c:825]: Since the next line prints ")", this ends up being "?)" which we've established is (in this file) a commonly-used truncation identifier.  Is it ok to keep this as-is?

File: print-dhcp6.c (hunk 5 above)
Comment (fenner) [reply print-dhcp6.c:816]: This falls through to the "remainder_len != 0" if below, which results in "?)", the common truncation marker in this file.

File: print-dhcp6.c (hunk 4 above)
Comment (fenner) [reply print-dhcp6.c:800]: I think these were copied from other examples in this file for values that are not NUL-terminated, e.g., there's no indication in https://datatracker.ietf.org/doc/html/rfc4833 that the timezone options are NUL-terminated but they use nd_printjnp (introduced in f6a0369d - that commit message even says that there is no NUL byte?).

I'm happy to fix up the newly-introduced calls.

**Comment by fenner** [comment]
I kept the fixup commits separate to make it clear - I will rebase everything to one commit once we come to agreement.

File: print-bootp.c (hunk 2 above)
Comment (infrastation) [reply print-bootp.c]: Yes, that's the idea, thank you for confirming. This way it would be easier to see what is on the wire, especially if the packet is not well-formed.

File: print-dhcp6.c (hunk 7 above)
Comment (infrastation) [reply print-dhcp6.c:825]: Yes, as discussed.

File: print-dhcp6.c (hunk 4 above)
Comment (infrastation) [reply print-dhcp6.c]: Regarding timezone options, Section 2 of RFC 4833 explicitly says:
> Note that they are NOT terminated by an ASCII NULL.

Commit f6a0369 perhaps could be refined through a separate follow-up.  Of course, there are many places for improvement in this source tree, some imperfections remain for a long time. That's a good reason not to introduce obvious imperfections with new code.

File: print-bootp.c [hunk 9]
```diff
@@ -993,6 +997,41 @@ rfc1048_print(netdissect_options *ndo,
 				break;
 			    }
 
+
+			case TAG_SZTP_REDIRECT:
+			    {
+				/* as per https://datatracker.ietf.org/doc/html/rfc8572#section-8.3
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+				 |	  uri-length		  |	     URI		  |
```
Comment (infrastation) [inline print-bootp.c:1005]: Cosmetic: tabs in the middle of this line distort the diagram.

File: print-bootp.c [hunk 10]
```diff
@@ -993,6 +997,41 @@ rfc1048_print(netdissect_options *ndo,
 				break;
 			    }
 
+
+			case TAG_SZTP_REDIRECT:
+			    {
```
Comment (infrastation) [inline print-bootp.c:1002]: Cosmetic: this pair of curly braces is not required.

**Comment by infrastation** [comment]
If anyone else was going to review these changes, please do it before long. I am going to have another look in a few days.

**Comment by fenner** [comment]
This push is simply a squash of the fixup commits, no new content.

**Comment by infrastation** [comment]
Thank you. I was getting there, but some other work unexpectedly got in the way. Will try again by tomorrow.

File: print-dhcp6.c [hunk 11]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
 			ND_PRINT(")");
 			break;
 
+		case DH6OPT_BOOTFILE_URL:
+			if (optlen < 2) {
```
Comment (infrastation) [inline print-dhcp6.c:794]: What constraint does this check implement? RFC 5970 Section 3.1 does not impose any requirements on `option-len` value. It does require `boot-file-url` to meet certain syntax requirements, is this related?

File: print-bootp.c [hunk 12]
```diff
@@ -993,6 +997,39 @@ rfc1048_print(netdissect_options *ndo,
 				break;
 			    }
 
+
+			case TAG_SZTP_REDIRECT:
+				/* as per https://datatracker.ietf.org/doc/html/rfc8572#section-8.3
+				 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-...-+-+-+-+-+-+-+
+				 |        uri-length              |          URI                  |
```
Comment (infrastation) [inline print-bootp.c:1004]: Almost there :-D

File: print-dhcp6.c [hunk 13]
```diff
@@ -784,6 +790,43 @@ dhcp6opt_print(netdissect_options *ndo,
...
+			ND_PRINT(")");
+			break;
+
+		case DH6OPT_SZTP_REDIRECT:
+		case DH6OPT_USER_CLASS:
+			ND_PRINT(" ");
+			tp = (const u_char *)(dh6o + 1);
+			first_list_value = TRUE;
+			remainder_len = optlen;
+			while (remainder_len > 2) {
```
Comment (infrastation) [inline print-dhcp6.c:810]: With `>= 2` this loop would make the final empty URI (if present) visible in the output.

**Comment by infrastation** [comment]
Thank you for waiting. The changes look safe. If you would like to fix a few remaining imperfections before the merge, please also add a change log entry.

File: print-dhcp6.c (hunk 11 above)
Comment (fenner) [reply print-dhcp6.c:794]: No, there is nothing special about 2 - a 4-octet url is just as likely to be too short as a 2-octet one, so I'll just remove the check and leave the validity understanding to the user.

Comment (infrastation) [reply print-dhcp6.c:794]: That's what I thought, thank you.

**Comment by infrastation** [comment]
Thank you!
